lib/rexml/parsers/baseparser.rb

```diff
@@ -460,15 +460,24 @@ def pull_event
                 @closed = tag
                 @nsstack.shift
               else
+                if @tags.empty? and @have_root
+                  raise ParseException.new("Malformed XML: Extra tag at the end of the document (got '<#{tag}')", @source)
+                end
                 @tags.push( tag )
               end
+              @have_root = true
               return [ :start_element, tag, attributes ]
             end
           else
             text = @source.read_until("<")
             if text.chomp!("<")
               @source.position -= "<".bytesize
             end
+            if @tags.empty? and @have_root
+              if text.strip != ""
```

Comment:
`strip` allocates a new string. Can we avoid it?
For example: `/\A\s*\z/.match?(text)`